components/more_direct_channels/index.js

```diff
@@ -49,7 +54,9 @@ function mapStateToProps(state, ownProps) {
     } else {
         users = getProfilesInCurrentTeam(state);
     }
-
+    const filteredDirectChannels = getChannelsWithUserProfiles(state).
+        filter((channel) => matchExistsInChannelProfiles(channel.profiles, searchTerm));
```

Comment:
We should memoize this filtering since doing it this way creates a new array for `filteredDirectChannels` each time `mapStateToProps` is called (which is constantly). One way to do that would be to use a selector or to use `memoizeResult` from `mattermost-redux/utils/helpers` to wrap this like
```javascript
const filterGroupChannels = memoizeResult((channels, term) => {
    return channels.filter((channel) => matchExistsInChannelProfiles(channel.profiles, term);
});

function mapStateToProps(state) {
    const filteredGroupChannels = filterGroupChannels(getChannelsWithUserProfiles(state), searchTerm);
}
```    

Also, this name is misleading since these are group channels, not direct channels, so that should be changed.